### ISSUE TITLE
KTOTrainer should work when actual batch size==1

### ISSUE THREAD

**Issue by starmpcc** [huggingface/trl#2554]
https://github.com/huggingface/trl/blob/edabe0a2d8fdd790319ce8862bb8e17336b85df1/trl/trainer/kto_trainer.py#L662-L665

This check was introduced in #2153
However, the KL logits were calculated by unlinking `prompt_input_ids` and `answer_input_ids`, which means the KL term is not equivalent to the reward term.
Accordingly, `KTOTrainer` should work when the actual batch size is 1.

Thank you!

**Comment by edbeeching** [comment]
Hi @starmpcc as the KTO KL term requires output ids for another sequence it necessitates that there is more than 1 sample in the batch.

**Comment by starmpcc** [comment]
Yes, but the KL dataset is already shuffled at the below codes, hence the check is not reqired
https://github.com/huggingface/trl/blob/edabe0a2d8fdd790319ce8862bb8e17336b85df1/trl/trainer/kto_trainer.py#L85-L93
